lib/src/int64.dart

```diff
@@ -59,10 +59,16 @@ class Int64 implements IntX {
   static Int64 _parseRadix(String s, int radix) {
     int i = 0;
     bool negative = false;
-    if (s[0] == '-') {
+    if (i < s.length && s[0] == '-') {
       negative = true;
       i++;
     }
+
+    // TODO(https://github.com/dart-lang/sdk/issues/38728): use (i >= s.length).
+    if (!(i < s.length)) {
```

Comment:
How about keeping the code the way it should be and adding this above it:

```dart
// ignore: invariant_booleans
```

(And keep the TODO comment.)